pkg/cloud/cloud_test.go

```diff
@@ -1711,28 +1745,39 @@ func TestCreateSnapshot(t *testing.T) {
 			mockEC2 := NewMockEC2API(mockCtrl)
 			c := newCloud(mockEC2)
 
-			ec2snapshot := &ec2.Snapshot{
-				SnapshotId: aws.String(tc.snapshotOptions.Tags[SnapshotNameTagKey]),
-				VolumeId:   aws.String("snap-test-volume"),
-				State:      aws.String("completed"),
-			}
-
 			ctx := context.Background()
-			mockEC2.EXPECT().CreateSnapshotWithContext(gomock.Any(), eqCreateSnapshotInput(tc.expInput)).Return(ec2snapshot, tc.expErr)
-			mockEC2.EXPECT().DescribeSnapshotsWithContext(gomock.Any(), gomock.Any()).Return(&ec2.DescribeSnapshotsOutput{Snapshots: []*ec2.Snapshot{ec2snapshot}}, nil).AnyTimes()
+
+			mockEC2.EXPECT().CreateSnapshot(gomock.Any(), gomock.Any(), gomock.Any()).Return(&ec2.CreateSnapshotOutput{
```

Comment:
If we're going to remove `eqCreateSnapshotInput`, is it worth using `DoAndReturn` to check the AWS SDK call contains the correct volume id/tags/description?

Reply:
Sure, added `DoAndReturn` validation.